configure.py

```diff
@@ -2218,8 +2218,8 @@ def write_build_file(f,
             command = clang --target=wasm32 --no-standard-libraries -Wl,--export-all -Wl,--no-entry $in -o $out
             description = C2WASM $out
         rule rust2wasm
-            command = cargo build --target=wasm32-wasi --example=$example --locked --manifest-path=test/resource/wasm/rust/Cargo.toml --target-dir=$builddir/wasm/ $
-                && wasm-opt -Oz $builddir/wasm/wasm32-wasi/debug/examples/$example.wasm -o $builddir/wasm/$example.wasm $
+            command = cargo build --target=wasm32-wasip1 --example=$example --locked --manifest-path=test/resource/wasm/rust/Cargo.toml --target-dir=$builddir/wasm/ $
```

Comment:
could you please cherry-pick related changes from #21184 so that the CMake-based build and document are in sync as well?

Reply:
Sure, unless it gets in first (in which case I'll rebase)

Reply:
Wait, cmake is already agnostic to the wasm flavor name.

Reply:
ahh, yes. #22320 was merged 3 days ago. but we still need at least:

```diff
diff --git a/install-dependencies.sh b/install-dependencies.sh
index e19e38a799..838d655db3 100755
--- a/install-dependencies.sh
+++ b/install-dependencies.sh
@@ -112,7 +112,7 @@ fedora_packages=(
     rust
     cargo
     rapidxml-devel
-    rust-std-static-wasm32-wasi
+    rust-std-static-wasm32-wasip1
     wabt
     binaryen
     lcov
```

we can fix `docs/cql/wasm.rst` later as a follow-up though.

Reply:
adjusted